docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -330,107 +756,124 @@ params.put("status", Status.Alive);
+=== Sorting
 
-Repository is a very popular pattern and can be very accurate for some use case, depending on
-the complexity of your needs.
+The following examples demonstrate different ways to sort.
 
-Whether you want to use the Entity based approach presented above or a more traditional Repository approach, it is up to you,
-Panache and Quarkus have you covered either way.
+* Methods that accept a query string also accept the following simplified query form:
++
+[source,java]
+----
+List<Person> persons = Person.list("order by name,birth");
+----
 
-If you lean towards using Repositories, you can get the exact same convenient methods injected in your Repository by making it
-implement `PanacheRepository`:
+* Methods that accept a query string accept an optional `Sort` parameter, which enables you to abstract your sorting:
+** Sort example 1
++
+[source,java]
+----
+List<Person> persons = Person.list(Sort.by("name").and("birth"));
+----
+** Sort example 2
++
 
 [source,java]
 ----
-@ApplicationScoped
-public class PersonRepository implements PanacheRepository<Person> {
+List<Person> persons = Person.list("status", Sort.by("name").and("birth"), Status.Alive);
+----
 
-   // put your custom logic here as instance methods
+The `Sort` class has many methods for adding columns and specifying sort direction.
 
-   public Person findByName(String name){
-       return find("name", name).firstResult();
-   }
+=== Query projection
 
-   public List<Person> findAlive(){
-       return list("status", Status.Alive);
-   }
+You can perform query projection with the `project(Class)` method on the `PanacheQuery` object that is returned by the `find()` methods. You can use it to restrict which fields will be returned by the database. Hibernate uses data transfer object (DTO) projection and generates a `SELECT` clause with the attributes from the projection class. This is also called _dynamic instantiation_ or _constructor expression_. For more information, see "The SELECT clause" section of the
+link:https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html[_Hibernate_User_Guide_].
 
-   public void deleteStefs(){
-       delete("name", "Stef");
-  }
-}
-----
+The projection class must be a valid Java Bean and have a constructor that contains all of its attributes. This constructor will be used to instantiate the projection DTO instead of using the entity class. This must be the only constructor of the class.
 
-Absolutely all the operations that are defined on `PanacheEntityBase` are available on your DAO, so using it
-is exactly the same except you need to inject it:
+The following example shows how to perform query projection:
 
 [source,java]
 ----
-@Inject
-PersonRepository personRepository;
+import io.quarkus.runtime.annotations.RegisterForReflection;
 
-@GET
-public long count(){
-    return personRepository.count();
+@RegisterForReflection
+public class PersonName {
+    public final String name; // <2>
```

Comment:
Looks like the `<2>` and `<3>` call-outs are missing here?

Reply:
I have the call out text in the main text. I have removed the call out reference from the example. Note that I have conditionalized the note about native executables.